plugins/network-elements/nuage-vsp/src/com/cloud/network/element/NuageVspElement.java

```diff
@@ -387,41 +400,39 @@ public boolean canEnableIndividualServices() {
 
     @Override
     public boolean destroy(Network network, ReservationContext context) throws ConcurrentOperationException, ResourceUnavailableException {
-        if (!canHandle(network, Service.Connectivity)) {
-            return false;
-        }
-
-        return true;
+        return canHandle(network, Service.Connectivity);
     }
 
     @Override
     public boolean verifyServicesCombination(Set<Service> services) {
```

Comment:
The method assumes that `services` is not `null`.  Therefore, please consider adding a `Preconditions.checkArgument` check to verify that `services` is not `null`